app/controllers/users_controller.rb

```diff
@@ -16,10 +16,24 @@ def destroy
     redirect_to users_url
   end
   
+  def edit_info
+    @user = User.find(params[:id])
+  end
+  
   def edit
     @user = User.find(params[:id])
   end
   
+  def update_info
```

Comment:
This looks like a duplicate of `update`.  Could we just add a little logic in `update` that will decide whether to render `edit_info` or just `edit` on failure?

Reply:
I don't think we can. It threw an error if I combine them. And the book needs 2 separate methods too.

Reply:
What sort of error?